packages/flutter/test/widgets/list_view_test.dart

```diff
@@ -18,6 +18,60 @@ class TestSliverChildListDelegate extends SliverChildListDelegate {
   }
 }
 
+class Alive extends StatefulWidget {
+  const Alive(this.alive, this.index);
+  final bool alive;
+  final int index;
+
+  @override
+  AliveState createState() => new AliveState();
+
+  @override
+  String toString({DiagnosticLevel minLevel}) => '$index $alive';
+}
+
+class AliveState extends State<Alive> with AutomaticKeepAliveClientMixin {
+  @override
+  bool get wantKeepAlive => widget.alive;
+
+  @override
+  Widget build(BuildContext context) =>
+     new Text('${widget.index}:$wantKeepAlive');
+}
+
+typedef WhetherToKeepAlive = bool Function(int);
+class _StatefulListView extends StatefulWidget {
+  const _StatefulListView(this.aliveCallback);
+
+  final WhetherToKeepAlive aliveCallback;
+  @override
+  _StatefulListViewState createState() => new _StatefulListViewState();
+}
+
+class _StatefulListViewState extends State<_StatefulListView> {
+  @override
+  Widget build(BuildContext context) {
+    return new GestureDetector(
+      onTap: () => setState,
```

Comment:
Why this? No state seems to be changing?

Reply:
Forces a rebuild. 

The issue is that on rebuild, an attempt at fixing this was dropping leading widgets with a keepAlive.  Rebuild here checks to make sure we're retaining those widgets.

Reply:
Makes sense. Probably worth adding a comment here explaining the strange usage of setstate.

Reply:
Done